src/thread_task.rs

```diff
@@ -207,43 +207,36 @@ unsafe fn insert_tasks(
     let mut current_block_34: u64;
     let mut maybe_t = tasks.head;
     while let Some(t) = maybe_t {
-        if tasks[t].type_0 == TaskType::TileEntropy {
-            if tasks[first].type_0 > TaskType::TileEntropy {
-                current_block_34 = 11174649648027449784;
-            } else if tasks[first].sby > tasks[t].sby {
-                current_block_34 = 11174649648027449784;
-            } else {
-                if tasks[first].sby < tasks[t].sby {
-                    insert_tasks_between(c, f, first, last, prev_t, Some(t), cond_signal);
-                    return;
-                }
-                current_block_34 = 15904375183555213903;
+        match (
+            tasks[t].type_0,
+            tasks[t].sby,
+            tasks[first].type_0,
+            tasks[first].sby,
+        ) {
+            // entropy coding precedes other steps
+            (TaskType::TileEntropy, _, f_type, _) if f_type > TaskType::TileEntropy => { /* continue */
             }
-        } else {
-            if tasks[first].type_0 == TaskType::TileEntropy {
+            // both are entropy
+            (_, t_sby, _, f_sby) if f_sby > t_sby => { /* continue */ }
+            (_, t_sby, _, f_sby) if f_sby < t_sby => {
```

Comment:
Like @randomPoison, I'm not sure this is correct.  Shouldn't the second case be if `t.type_0 != TaskType::TileEntropy && first.type_0 == TaskType::TileEntropy`?

Note that if we get the logic wrong here, I don't think we get a test failure, just slower multithreading, so we need to be careful.

I think if you do my suggestion above of defining `t` and `first` variables and then `match (t.type_0, first.type_0)`ing, it'll be easier to understand the logic.